rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -321,3 +337,115 @@ def get_python_type(type_):
         return 'str'
 
     assert False, "unknown type '%s'" % type_
+
+
+def get_type_annotation_constant(constant: Constant) -> str:
+    return get_type_annotation_constant_default(constant, constant.value)
+
+
+def get_type_annotation_default(member: Member) -> str:
+    default = member.get_annotation_value('default')
+    assert isinstance(default, dict), 'Only default types are dictionary'
+    return get_type_annotation_constant_default(member, default['value'])
+
+
+def get_type_annotation_constant_default(constant: Union[Constant, Member],
+                                         value: Union[str, int, float, bool]) -> str:
+    """From a constant/member return type annotations for constant/default values."""
+    type_ = constant.type
+
+    if isinstance(type_, AbstractNestedType):
+        type_ = type_.value_type
+
+    python_type = get_python_type(type_)
+
+    if isinstance(constant.type, AbstractNestedType) and isinstance(type_, BasicType) and \
+       type_.typename in SPECIAL_NESTED_BASIC_TYPES:
```

Comment:
Minor comment, feel free to ignore:

Personally, I'm a fan of `if (...):` for readability (I think the `and` goes at the end of the previous line and not at the beginning of the next line, but I'm not 100% sure):

```suggestion
    if (
        isinstance(constant.type, AbstractNestedType) and isinstance(type_, BasicType) and
        type_.typename in SPECIAL_NESTED_BASIC_TYPES
    ):
```